engine-tests/src/test/java/org/terasology/math/BlockRegionTest.java

```diff
@@ -0,0 +1,165 @@
+        List<Vector3i> expectedSize = Arrays.asList(new Vector3i(1, 1, 1), new Vector3i(3, 3, 3), new Vector3i(8, 5, 2));
+        List<Vector3i> max = Arrays.asList(new Vector3i(1, 1, 1), new Vector3i(4, 4, 4), new Vector3i(11, 9, 7));
+        for (int i = 0; i < mins.size(); ++i) {
+            BlockRegion region = new BlockRegion().union(mins.get(i)).union(max.get(i));
+            assertEquals(mins.get(i), region.getMin(new Vector3i()));
+            assertEquals(max.get(i), region.getMax(new Vector3i()));
+            assertEquals(expectedSize.get(i), region.getSize(new Vector3i()));
+        }
+    }
+
+    @Test
+    public void testCreateRegionWithBounds() {
+        BlockRegion expectedRegion = new BlockRegion(new Vector3i(-2, 4, -16), new Vector3i(5, 108, 1));
+        List<Vector3i> vec1 = Arrays.asList(new Vector3i(-2, 4, -16), new Vector3i(4, 4, -16), new Vector3i(-2, 107, -16), new Vector3i(-2, 4, 0),
+            new Vector3i(4, 107, -16), new Vector3i(4, 4, 0), new Vector3i(-2, 107, 0), new Vector3i(4, 107, 0));
+        List<Vector3i> vec2 = Arrays.asList(new Vector3i(4, 107, 0), new Vector3i(-2, 107, 0), new Vector3i(4, 4, 0), new Vector3i(4, 107, -16),
+            new Vector3i(-2, 4, 0), new Vector3i(-2, 107, -16), new Vector3i(4, 4, -16), new Vector3i(-2, 4, -16));
+        for (int i = 0; i < vec1.size(); ++i) {
+            assertEquals(expectedRegion, new BlockRegion().unionBlock(vec1.get(i)).unionBlock(vec2.get(i)));
+        }
+    }
+
+    @Test
+    public void testRegionEmptyIfMaxLessThanMin() {
+        BlockRegion region = new BlockRegion(new Vector3i(0, 0, 0), new Vector3i(-1, 0, 0));
+        assertFalse(region.isValid());
+    }
+
+    @Test
+    public void testRegionEmptyIfSizeZeroOrLess() {
+        BlockRegion region = new BlockRegion().union(new Vector3i(1, 1, 1)).setSize(new Vector3i(0, 1, 1));
+        assertFalse(region.isValid());
+        region = new BlockRegion().union(new Vector3i(1, 1, 1)).setSize(new Vector3i(1, -1, 1));
+        assertFalse(region.isValid());
+    }
+
+    @Test
+    public void testIterateRegion() {
+        Vector3i min = new Vector3i(2, 5, 7);
+        Vector3i max = new Vector3i(10, 11, 12);
+        BlockRegion region = new BlockRegion().unionBlock(min).unionBlock(max);
+
+        Set<Vector3ic> expected = Sets.newHashSet();
+        for (int x = min.x; x <= max.x; ++x) {
+            for (int y = min.y; y <= max.y; ++y) {
+                for (int z = min.z; z <= max.z; ++z) {
+                    expected.add(new Vector3i(x, y, z));
+                }
+            }
+        }
+
+
+        for (Vector3ic pos : BlockRegionIterable.region(region).build()) {
+            assertTrue(expected.contains(pos), "unexpected position: " + pos);
+            expected.remove(pos);
+        }
+
+        assertEquals(0, expected.size(), "All vectors provided");
+    }
+
+    @Test
+    public void testSimpleIntersect() {
```

Comment:
The usage of `unionBlock()` in these intersection tests is absolutely irrelevant. You should use the min-max constructor and adjust the expected values where necessary.